### PR TITLE
{chem}[foss/2019a] Libint v2.0.3

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
nic426 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/71ea24703c542adae0ff89a351aeef0a for a full test report.

**Comment by vanzod** [comment]
@smoors Is there a specific reason you are pushing an "old" version to the `foss-2019a` toolchain? With `iimpi-2019a` we have v2.5.0 so it may make sense to mirror that.
For the same reason, can you also build it with the `gompi` toolchain?

**Comment by smoors** [comment]
@vanzod I need the 2.0.3 version for building `horton` with the 2019a toolchain generation. I agree we should have a v2.5.0 version with GCC as well, I will make it.
However I am not so sure about moving from `foss` to `gompi`. From grepping the sources of v2.5.0 it looks like BLAS and LAPACK are used if available. In v2.0.3 LAPACK seems to be used if available.

**Comment by Micket** [comment]
@smoors Apart from some mkl_cblas in profile.cc and test i see no mention of any blas/lapack headers. Couldn't see anything in any of the build scripts either.

Maybe via Eigen? And even if we haven't been doing so, perhaps we should be defining EIGEN_USE_BLAS + EIGEN_USE_LAPACKE  or EIGEN_MKL_USEALL.
I'm not sure how eigen performs without these, but, at least for MKL it probably makes a difference. Libint does use axpy and dot_products. 
But, it also uses a lot of builtin vector intrinsics, so, it might not matter at all for Libint.

I tried a test build for this, and, it fails to compile for me. __mm_fmadd_pd isn't in emmintrin.h for me.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.6.1810, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/bde135725fbff5ac25e50feab274bff2 for a full test report.

**Comment by smoors** [comment]
@Micket I found some code using `#if HAVE_LAPACK`, but indeed `HAVE_LAPACK` is not defined anywhere. I am going to trust you wrt the performance (as I don't have the time to do benchmarking), so will change the toolchain to `gompi`.

for the failed report, see the comment in the easyconfig:
`add 'optarch': 'mavx2' for skylake cpus`

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
hebbe-s1.int.private - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz, Python 2.7.14
See https://gist.github.com/5a16b7d0b9c024510398221c78a69b96 for a full test report.

**Comment by Micket** [comment]
@smoors Noone should trust me as an authority here! This is just my thoughts after skimming the code. I'm not against moving it to toolchain level if it improves performance. But my impression is that it would be fine to stick to gompi.

Also, it's not just Skylake, Haswell as well (see above).
I'm also a bit surprised, as your test report was using Skylake.
AVX2 has been around for a while, shouldn't we consider adding it on by default then?

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
login1.cerberus.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ec4c6d9eb2ce994c0ebf2f9707c94a50 for a full test report.

**Comment by smoors** [comment]
@Micket I've made the changes as you suggested.
(my first test report was also with -mavx2, that's why it succeeded.)

File: easybuild/easyconfigs/l/Libint/Libint-2.5.0-gompi-2019a.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+source_urls = ['https://github.com/evaleev/libint/archive']
+sources = ['v%(version)s.tar.gz']
+checksums = ['e57bb4546a6702fdaa570ad6607712f31903ed4618f051150979a31a038ce960']
+
+builddependencies = [
+    ('Autotools', '20180311'),
+    ('GMP', '6.1.2'),
+    ('Boost', '1.70.0'),
+    ('Eigen', '3.3.7', '', True),
+    ('Python', '2.7.15'),
```
Comment (Micket) [inline easybuild/easyconfigs/l/Libint/Libint-2.5.0-gompi-2019a.eb]: Can it use Python3 when building?

Comment (smoors) [reply easybuild/easyconfigs/l/Libint/Libint-2.5.0-gompi-2019a.eb]: @Micket it's not Python3 compatible (but you could make it compatible without too much effort if you really wanted).

Comment (Micket) [reply easybuild/easyconfigs/l/Libint/Libint-2.5.0-gompi-2019a.eb]: @smoors I started testbuilding it myself, and indeed, it seems like (at least) one of the tests have an uses python2 stuff. execfile. Libint itself happily uses python3 for the rest though. I will report a bug to them, and maybe they will fix it for the next release.

I think we can just stick with python2 for now.

**Comment by Micket** [comment]
Going in, thanks @smoors!

**Pull request merged**